File: be/src/main/java/com/project/popupmarket/service/receipts/StagingPaymentRedisService.java
package com.project.popupmarket.service.receipts;

import com.project.popupmarket.entity.StagingPayment;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Service;

import java.util.concurrent.TimeUnit;

@Service
@RequiredArgsConstructor
public class StagingPaymentRedisService {
    private final RedisTemplate<String, Object> template;

    private String generateKey(String orderId) {
        return "staging:" + orderId;
    }

    public void save(String id, StagingPayment stagingPayment) {
        String key = generateKey(id);
        template.opsForValue().set(key, stagingPayment, 5L, TimeUnit.MINUTES);
    }

    public StagingPayment find(String id) {
        String key = generateKey(id);
        return (StagingPayment) template.opsForValue().get(key);
    }

    public void delete(String id) {
        String key = generateKey(id);
        template.delete(key);
    }
}

File: be/src/main/java/com/project/popupmarket/service/receipts/StagingPaymentRedisService.kt
package com.project.popupmarket.service.receipts

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import com.project.popupmarket.entity.StagingPayment
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Service
import java.util.concurrent.TimeUnit

@Service
class StagingPaymentRedisService(
    private val template: RedisTemplate<String, Any>,
    private val objectMapper: ObjectMapper
) {
    private fun generateKey(orderId: String): String = "staging:$orderId"

    fun save(id: String, payment: StagingPayment) {
        val key = generateKey(id)
        template.opsForValue().set(key, payment, 10L, TimeUnit.MINUTES)
    }

    fun find(id: String): StagingPayment? {
        val key = generateKey(id)
        val rawData = template.opsForValue().get(key) ?: return null

        val jsonString = objectMapper.writeValueAsString(rawData)
        return objectMapper.readValue(jsonString, object : TypeReference<StagingPayment>() {})
    }

    fun delete(id: String) {
        val key = generateKey(id)
        template.delete(key)
    }
}
